### ISSUE TITLE
HTTP requests not valid

### ISSUE THREAD

**Issue by oeb25** [Phrohdoh/firebase-rs#10]
Seems like my pull request was a little to hasty, I'm getting HTTP errors from what seems to be hyper, but haven't been able to track down the exact issue just yet.

Will look into it, and see if I can identify the problem. Sorry for messing up :disappointed:

The error looks like the following

```rust
NetworkErr(
    Io(
        Error {
            repr: Custom(
                Custom {
                    kind: InvalidInput,
                    error: StringError(
                        "Invalid scheme for Http"
                    )
                }
            )
        }
    )
)
```

**Comment by oeb25** [comment]
The issue occurs [here](https://github.com/Phrohdoh/firebase-rs/blob/master/src/lib.rs#L354), and I'm pretty sure it's related to hyper dropping OpenSSL support out of the box, see [this commit](https://github.com/hyperium/hyper/commit/2f48612c7e141a9d612d7cb9d524b2f460561f56).

Will try to find a solution for this.

**Comment by phrohdoh** [comment]
This was, hopefully fully, resolved in 2db8ae1d9b7d17f0651e1c3236f6e5bac898f4a8.
